tests/internal/test_process_graph_visitor.py

```diff
@@ -126,21 +126,21 @@ def test_visit_nodes_array():
 def test_visit_array_with_dereferenced_nodes():
     graph = {
         'arrayelement1': {
-            'arguments': {'data': {'from_argument': 'data'}, 'index': 2},
+            'arguments': {'data': {'from_argument': 'data'}, 'index': {"from_parameter":"index"}},
             'process_id': 'array_element',
             'result': False
         },
         'product1': {
             'process_id': 'product',
-            'arguments': {'data': [{'from_node': 'arrayelement1'}, -1]},
+            'arguments': {'data': [{'from_node': 'arrayelement1'}, -1,{"from_parameter":"multiplier"}]},
             'result': True
         }
     }
     top = ProcessGraphVisitor.dereference_from_node_arguments(graph)
     dereferenced = graph[top]
     assert dereferenced["arguments"]["data"][0]["arguments"]["data"]["from_argument"] == "data"
 
-    visitor = ProcessGraphVisitor()
+    visitor = ProcessGraphVisitor(parameters={"index":2,"multiplier":3})
```

Comment:
as noted above: `ProcessGraphVisitor` does not have a `parameter` argument/attribute in this PR yet